src/java.base/share/classes/jdk/internal/classfile/impl/SplitConstantPool.java

```diff
@@ -0,0 +1,613 @@
+ * questions.
+ */
+package jdk.internal.classfile.impl;
+
+import java.lang.constant.ConstantDesc;
+import java.lang.constant.MethodTypeDesc;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+import jdk.internal.classfile.Attribute;
+import jdk.internal.classfile.Attributes;
+import jdk.internal.classfile.ClassReader;
+import jdk.internal.classfile.Classfile;
+import jdk.internal.classfile.constantpool.ClassEntry;
+import jdk.internal.classfile.constantpool.ConstantDynamicEntry;
+import jdk.internal.classfile.constantpool.ConstantPoolBuilder;
+import jdk.internal.classfile.constantpool.ConstantPool;
+import jdk.internal.classfile.BootstrapMethodEntry;
+import jdk.internal.classfile.BufWriter;
+import jdk.internal.classfile.attribute.BootstrapMethodsAttribute;
+import jdk.internal.classfile.constantpool.DoubleEntry;
+import jdk.internal.classfile.constantpool.FieldRefEntry;
+import jdk.internal.classfile.constantpool.FloatEntry;
+import jdk.internal.classfile.constantpool.IntegerEntry;
+import jdk.internal.classfile.constantpool.InterfaceMethodRefEntry;
+import jdk.internal.classfile.constantpool.InvokeDynamicEntry;
+import jdk.internal.classfile.constantpool.LoadableConstantEntry;
+import jdk.internal.classfile.constantpool.LongEntry;
+import jdk.internal.classfile.constantpool.MemberRefEntry;
+import jdk.internal.classfile.constantpool.MethodHandleEntry;
+import jdk.internal.classfile.constantpool.MethodRefEntry;
+import jdk.internal.classfile.constantpool.MethodTypeEntry;
+import jdk.internal.classfile.constantpool.ModuleEntry;
+import jdk.internal.classfile.constantpool.NameAndTypeEntry;
+import jdk.internal.classfile.constantpool.PackageEntry;
+import jdk.internal.classfile.constantpool.PoolEntry;
+import jdk.internal.classfile.constantpool.StringEntry;
+import jdk.internal.classfile.constantpool.Utf8Entry;
+
+import static jdk.internal.classfile.Classfile.TAG_CLASS;
+import static jdk.internal.classfile.Classfile.TAG_CONSTANTDYNAMIC;
+import static jdk.internal.classfile.Classfile.TAG_DOUBLE;
+import static jdk.internal.classfile.Classfile.TAG_FIELDREF;
+import static jdk.internal.classfile.Classfile.TAG_FLOAT;
+import static jdk.internal.classfile.Classfile.TAG_INTEGER;
+import static jdk.internal.classfile.Classfile.TAG_INTERFACEMETHODREF;
+import static jdk.internal.classfile.Classfile.TAG_INVOKEDYNAMIC;
+import static jdk.internal.classfile.Classfile.TAG_LONG;
+import static jdk.internal.classfile.Classfile.TAG_METHODHANDLE;
+import static jdk.internal.classfile.Classfile.TAG_METHODREF;
+import static jdk.internal.classfile.Classfile.TAG_METHODTYPE;
+import static jdk.internal.classfile.Classfile.TAG_MODULE;
+import static jdk.internal.classfile.Classfile.TAG_NAMEANDTYPE;
+import static jdk.internal.classfile.Classfile.TAG_PACKAGE;
+import static jdk.internal.classfile.Classfile.TAG_STRING;
+
+/**
+ * ConstantPool.
+ */
+public final class SplitConstantPool implements ConstantPoolBuilder {
```

Comment:
Not sure the "Split" prefix carries any meaning? (perhaps a leftover from previous iterations?)

Reply:
The "Split" represent a division between the original read-only constant pool and the new part with additional CP entries.